### PR TITLE
effects: Replace old-style Qt connect() calls

### PR DESCRIPTION
**Pull request by Holzhaus**

### CODE REVIEW

**Comment by Be-ing** [comment]
Please don't make more merge conflicts for #2618 just to make a linter happy.

**Comment by uklotzde** [comment]
@Be-ing Approval to merge this?

The integration of clazy will push code quality onto a new level and helps to detect issues upfront before manually reviewing PRs. It will save us from a lot of tedious review work and prevent bugs. I am so glad that Jan has started this.

File: src/effects/effectchainslot.h [hunk 1]
```diff
@@ -108,7 +108,7 @@ class EffectChainSlot : public QObject {
     void slotControlClear(double v);
     void slotControlChainEnabled(double v);
     void slotControlChainMix(double v);
-    void slotControlChainSuperParameter(double v, bool force = false);
+    void slotControlChainSuperParameter(double v, bool force);
```
Comment (uklotzde) [inline src/effects/effectchainslot.h:111]: Why did you decide keep the default parameter shouldEmit = true in slotChainEffectChanged() although you pass `true` explicitly now? Default parameters in slots should be avoided.

**Review by uklotzde — CHANGES_REQUESTED** [review]
TODO: Remove default parameter from slotChainEffectChanged()

**Comment by uklotzde** [comment]
@Be-ing Conflicts with #2618 should be easy to resolve. The removal of default parameters from slots is the only notable exceptions.

**Comment by Be-ing** [comment]
> TODO: Remove default parameter from slotChainEffectChanged()

Adding TODOs in code that has already been completely rewritten is pointless.

**Comment by Be-ing** [comment]
> @Be-ing Approval to merge this?

Okay. I'm just grumpy about having to keep resolving merge conflicts.

**Comment by uklotzde** [comment]
Maybe it is better we resolve those merge conflicts now instead of later whenever we need to merge 2.3 into main?

We need to ensure that no major effect changes are required for 2.3 before merging #2618, including skins. But then I recommend to merge #2618 early for thorough testing!

**Comment by Be-ing** [comment]
> Maybe it is better we resolve those merge conflicts now instead of later whenever we need to merge 2.3 into main?

I don't understand how that would be possible.

**Review by uklotzde — APPROVED** [review]
LGTM

**Comment by Holzhaus** [comment]
I could take care of the merge conflicts in #2618, but it already conflicts and I have no idea about the skin conflicts.

**Comment by uklotzde** [comment]
No need to wait for the redundant SCons build. LGTM

**Pull request merged**